pkg/registry/apps/deployment/storage/storage_test.go

```diff
@@ -296,7 +296,7 @@ func TestStatusUpdate(t *testing.T) {
 		},
 	}
 
-	if _, _, err := storage.Status.Update(ctx, update.Name, rest.DefaultUpdatedObjectInfo(&update), rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc); err != nil {
+	if _, _, err := storage.Status.Update(ctx, update.Name, rest.DefaultUpdatedObjectInfo(&update), rest.DefaultUpdateConfig(rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, false), &metav1.UpdateOptions{}); err != nil {
```

Comment:
@jennybuckley I haven't taken another look on your PR, yet. From the changes here: wouldn't a constant like `UpdateNoDryRun` (and a `type UpdateMode bool` type) be much easier to read? This `false` or even a `true` value looks far too innocent.